Issue-76-78: Genre & People Detail Bugs

glnygl opened the pull request:
Issue-76:  NavigationLink without destination code fixed that issue

![Simulator Screen Recording - iPhone 15 Pro - 2024-01-16 at 12 49 59](https://github.com/Dimillian/MovieSwiftUI/assets/26183230/339a5a19-4b40-4a60-8e6d-f02a022acce2)

Issue-78: Remove animations fixed that issue

![Simulator Screen Recording - iPhone 15 Pro - 2024-01-17 at 18 29 17](https://github.com/Dimillian/MovieSwiftUI/assets/26183230/b73f576d-f988-49e4-9a79-bfe815029244)

+ Fixed thread warning

glnymigrosone committed:
Issue-76: Fix genre detail pop back bug

glnymigrosone committed:
Issue-78: Remove animations from PeopleDetail body

Kiyzer commented:
Hello everybody,
first of all, congratulations for this really great project..
same issue when you are in the Genres list, you have the list of films according to genre but as soon as you click on a film it returns to the list of films of the Genre.
Same when you are looking for a Film with a comment, as soon as you click on the comment it is not displayed and returns to the FILM  detail page.
I hope my descriptions of the bugs isn't confusing and that this hasn't already been covered.
Ogun

glnygl commented:
> Hello everybody, first of all, congratulations for this really great project.. same issue when you are in the Genres list, you have the list of films according to genre but as soon as you click on a film it returns to the list of films of the Genre. Same when you are looking for a Film with a comment, as soon as you click on the comment it is not displayed and returns to the FILM detail page. I hope my descriptions of the bugs isn't confusing and that this hasn't already been covered. Ogun

Hi Ogun, I think I fixed these bugs that you mentioned with this pr. This pop back problem is caused by different navigation code. You can also check this fix in your simulator.  If still there is a problem maybe you can add a screetshot of video.

Kiyzer commented:
@glnygl
Hi,
Gülenay,
yes I see your corrections and did . It solved the bugs in Genres Navigations in de Movie details and the cast details but it seems persist when you are in genres List  and Review.

https://github.com/Dimillian/MovieSwiftUI/assets/96753216/e0dedfb1-abec-4bcb-9c7d-fa35895a6749

and the same issue when you tap on Reviews.
I see the same bug when you create a custom list, and you want to  check the movie details pop back
Ogün ;)

glnygl commented:
> @glnygl Hi, Gülenay, yes I see your corrections and did . It solved the bugs in Genres Navigations in de Movie details and the cast details but it seems persist when you are in genres List and Review.
> 
>  RPReplay_Final1707084577.MP4 
> and the same issue when you tap on Reviews. I see the same bug when you create a custom list, and you want to check the movie details pop back Ogün ;)

Hi Ogun, now i see the problem. Video helped a lot thanks for this. After debug I realized that there is a problem in store. When I comment out "fetchMovieDetails" func in MovieDetail view it works properly. But i didn't understand why dispatch func cause this problem. Im still on it. If you find a way to solve this bug please inform me. Im really curious about it. 😅

glnygl committed:
Fix: Thread warning issue with receive func and switch thread to main
:rotating_light: Warning description: Publishing changes from within view updates is not allowed, this will cause undefined behavior.
